src/core/meshcop/dataset_manager.cpp

```diff
@@ -199,8 +199,7 @@ Error DatasetManager::ApplyConfiguration(const Dataset &aDataset) const
 
             if (error != kErrorNone)
             {
-                LogWarn("ApplyConfiguration() Failed to set channel to %d (%s)", channel, ErrorToString(error));
-                ExitNow();
+                LogWarn("Failed to set channel to %u when applying dataset: %s", channel, ErrorToString(error));
```

Comment:
What about `LogCrit()` here? LogWarn is normally pretty mild like e.g. failing to send a particular message (which can be easily retried) or not able to store some extra items.  While this error is critical: the dataset defines a channel that the device can't support, and that's pretty final. (Typically hardcoded in the platform in the supported-channels mask.) So the entire Thread device can't operate anymore as part of the network, on the intended channel.

Reply:
Thanks @EskoDijk . I think using `LogCrit()` can be useful. Updated in new push.